unblob/logging.py

```diff
@@ -31,16 +31,22 @@ def _format_message(value: Any, extract_root: Path) -> Any:
 
     elif isinstance(value, Path):
         try:
-            return str(value.relative_to(extract_root))
+            return (
+                value.relative_to(extract_root)
+                .as_posix()
+                .encode("utf-8", errors="surrogateescape")
+            )
         except ValueError:
             # original files given to unblob may not be relative to extract_root
-            return str(value)
+            return value.as_posix().encode("utf-8", errors="surrogateescape")
 
     elif isinstance(value, Instance):
         return dumpstruct(value, output="string")
 
     elif isinstance(value, int):
         return format_hex(value)
+    elif isinstance(value, str):
+        return value.encode("utf-8", errors="surrogateescape")
```

Comment:
There was a smarter way of doing this. I need it to properly encode the stdout/stderr in case of errors, so I moved the encoding to the logger call in extractor (see https://github.com/IoT-Inspector/unblob/pull/194/files#diff-9f4b3ab3c14d83d69adeba706796c94983b997912f147d92738229b69fc11a16R69) and got rid of it in logging.py

Reply:
Ok, makes sense! Just fix the logging tests by removing your `str`->`byte` conversion on the non-path expected values, and we are good to go :smile:

Reply:
Ahah, done. I blame it on writing code before the week-end.